src/drivers/general/queries/RoutineFunctionTypeNotExpected/driver_snippet.c

```diff
@@ -41,22 +57,50 @@ char functionCallThatUsesFunctionPointer2(char a, funcCall functionPointer, char
+    funcCall2 f_bad_params2 = (funcCall2)voidFunctionToCall;
+
     funcCall f2 = &intFunctionToCall; // funcCall type specifies a void return type, but this is a function pointer to intFunctionToCall which returns an int
     funcCallInt f3 = &intFunctionToCall;
+
     functionCallThatUsesFunctionPointer(f1);       // pass
     functionCallThatUsesFunctionPointer(fun_ptr2); // pass
 
-    functionCallThatUsesFunctionPointer(fun_ptr1); // fail because fun_ptr1 is a function pointer to intFunctionToCall which returns an int
-    functionCallThatUsesFunctionPointer(f2);       // This SHOULD fail because f2 is a function pointer to intFunctionToCall which returns an int, but it is declared with a funcCall type which specifies a void return type so it passes
-    functionCallThatUsesFunctionPointer(f3);       // fail because f3 is a function pointer to intFunctionToCall which returns an int
+    functionCallThatUsesFunctionPointer(fun_ptr1);            // fail because fun_ptr1 is a function pointer to intFunctionToCall which returns an int
+    functionCallThatUsesFunctionPointer(f2);                  // This SHOULD fail because f2 is a function pointer to intFunctionToCall which returns an int, but it is declared with a funcCall type which specifies a void return type so it passes
+    functionCallThatUsesFunctionPointer(f3);                  // fail because f3 is a function pointer to intFunctionToCall which returns an int
     functionCallThatUsesFunctionPointer(&voidFunctionToCall); // pass
     functionCallThatUsesFunctionPointer(&intFunctionToCall);  // fail because this intFunctionToCall returns an int
 
     functionCallThatUsesFunctionPointer(voidFunctionToCall); // pass
-    functionCallThatUsesFunctionPointer(intFunctionToCall);  // fail because this intFunctionToCall returns an int
+    functionCallThatUsesFunctionPointer(intFunctionToCall);  // fail because intFunctionToCall returns an int
+
+    functionCallThatUsesFunctionPointerLong((funcCallLong)intFunctionToCall); // should pass because the int type is cast to long type
 
     functionCallThatUsesFunctionPointer2('a', intFunctionToCall, 'b'); // fail because this intFunctionToCall returns an int
+
+    // Check both return values and parameters
+    functionCallThatUsesFunctionPointer3(f_with_params);                // pass because voidFunctionToCallWithParams matches the parameters expected by funcCall2
+    functionCallThatUsesFunctionPointer3(voidFunctionToCallWithParams); // pass because voidFunctionToCallWithParams matches the parameters expected by funcCall2
+
+    functionCallThatUsesFunctionPointer3(f1);                 // fail because voidFunctionToCall has no parameters. NOTE the compiler warns about this
+    functionCallThatUsesFunctionPointer3(f_bad_params);       // fail because voidFunctionToCall has no parameters. The compiler does not warn about this BUT it warns about the intial asignemnt of f_bad_params
```

Comment:
I don't see SARIF results at lines 103 or 104.  Is this intentional? If so, please make this more clear in the comments.  

Also, typo: "intial asignemnt" should be "initial assignment".